core/server/worker/src/test/java/alluxio/worker/block/BlockMasterClientTest.java

```diff
@@ -77,4 +121,314 @@ public void convertBlockListMapToProtoMergeDirsInSameTier() {
+    final Map<String, List<String>> lostStorage = ImmutableMap.of(
+        "MEM",
+        ImmutableList.of("/tmp/lost")
+    );
+    final List<Metric> metrics = ImmutableList.of();
+
+    Channel mockChannel = createMockService(
+        new BlockMasterWorkerServiceGrpc.BlockMasterWorkerServiceImplBase() {
+          @Override
+          public void blockHeartbeat(BlockHeartbeatPRequest request,
+                                     StreamObserver<BlockHeartbeatPResponse> responseObserver) {
+            responseObserver.onNext(
+                BlockHeartbeatPResponse.newBuilder().setCommand(Command.newBuilder().setCommandType(
+                    CommandType.Nothing)).build()
+            );
+            responseObserver.onCompleted();
+          }
+        });
+
+    BlockMasterClient client = new MockStubBlockMasterClient(
+        MasterClientContext.newBuilder(ClientContext.create(mConf)).build(),
+        mockChannel
+    );
+
+    assertEquals(CommandType.Nothing, client.heartbeat(
+        workerId,
+        capacityBytesOnTiers,
+        usedBytesOnTiers,
+        removedBlocks,
+        addedBlocks,
+        lostStorage,
+        metrics).getCommandType());
+  }
+
+  @Test(expected = FailedToAcquireRegisterLeaseException.class)
+  public void testAcquireRegisterLeaseFailure() throws Exception {
+    testAcquireRegisterLease(false);
+  }
+
+  @Test
+  public void testAcquireRegisterLeaseSuccess() throws Exception {
+    testAcquireRegisterLease(true);
+  }
+
+  @Test
+  public void testRegister() throws Exception {
+    register(false);
+  }
+
+  @Test
+  public void testRegisterStream() throws Exception {
+    register(true);
+  }
+
+  private void testAcquireRegisterLease(boolean expectedSuccess) throws Exception {
+    Channel mockChannel = createMockService(
+        new BlockMasterWorkerServiceGrpc.BlockMasterWorkerServiceImplBase() {
+          @Override
+          public void requestRegisterLease(
+              GetRegisterLeasePRequest request,
+              StreamObserver<GetRegisterLeasePResponse> responseObserver) {
+            responseObserver.onNext(
+                GetRegisterLeasePResponse.newBuilder().setAllowed(expectedSuccess).build());
+            responseObserver.onCompleted();
+          }
+        });
+
+    BlockMasterClient client = new MockStubBlockMasterClient(
+        MasterClientContext.newBuilder(ClientContext.create(mConf)).build(),
+        mockChannel
+    );
+
+    client.acquireRegisterLeaseWithBackoff(
+        0L,
+        1,
+        ExponentialTimeBoundedRetry
+            .builder()
+            .withMaxDuration(Duration.of(500, ChronoUnit.MILLIS))
+            .withInitialSleep(Duration.of(0, ChronoUnit.MILLIS))
+            .withMaxSleep(Duration.of(100, ChronoUnit.MILLIS))
+            .withSkipInitialSleep()
+            .build());
+  }
+
+  public void register(boolean stream) throws Exception {
+    final long workerId = 0L;
```

Comment:
Use a non zero value here, as the default value for the worker id in `RegisterWorkerPRequest` is 0, if that field is not set. Otherwise, should the test subject forget to set the correct worker ID in the request and leave it to default, this test would fail to catch that.

Reply:
Got it. Fixed